LCM/scripts/OperationStatusUtility.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+from datetime   import datetime
+from imp        import load_source
+from os         import chmod, mkdir, stat
+from os.path    import dirname, join, isdir, isfile, realpath
+from re         import search
+from sys        import version_info
+
+if hasattr(datetime, 'strptime'):
+    # Only available in this module in Python 2.5+
+    strptime = datetime.strptime
+else:
+    # Only for Python 2.4
+    import time
+    strptime = lambda date_string, format: datetime(*(time.strptime(date_string, format)[0:6]))
+
+pathToCurrentScript = realpath(__file__)
+pathToCommonScriptsFolder = dirname(pathToCurrentScript)
+
+helperLibPath = join(pathToCommonScriptsFolder, 'helperlib.py')
+helperlib = load_source('helperlib', helperLibPath)
+
+def get_current_time_no_ms():
+    currentTimestamp = get_current_timestamp()
+    currentTimeNoMS = strptime(currentTimestamp, '%Y/%m/%d %H:%M:%S')
+    return currentTimeNoMS
+
+def get_current_timestamp():
+    currentDateTime = datetime.now()
+    currentDateTimeFormattedString = datetime.strftime(currentDateTime, "%Y/%m/%d %H:%M:%S")
+    return currentDateTimeFormattedString
+
+def get_permission_in_oct(threeDigitString):
+    # Python 2.4 and 3 recognize different formats for octal
+    if version_info >= (3, 0):
+        strMode = "0o" + threeDigitString
+    else:
+        strMode = "0" + threeDigitString
+
+    octMode = int(strMode, base = 8)
+    return octMode
+
+def ensure_file_permissions(filePath, permissionString):
+    desiredPermission = get_permission_in_oct(permissionString)
+
+    # Ensure that the file has the desired permissions if it exists
+    if (isfile(filePath)):
+        filePermission = oct(stat(filePath).st_mode & desiredPermission)
+        if (not (filePermission == desiredPermission)):
+            chmod(filePath, desiredPermission)
+
+def ensure_directory_permissions(directoryPath, permissionString):
+    desiredPermission = get_permission_in_oct(permissionString)
+
+    # Ensure that the file has the desired permissions if it exists
+    if (isdir(directoryPath)):
+        directoryPermission = oct(stat(directoryPath).st_mode & desiredPermission)
+        if (not (directoryPermission == desiredPermission)):
+            chmod(directoryPath, desiredPermission)
+
+def get_log_since_datetime(startDateTime):
+    logSinceDateTime = ''
+    foundStartDateTime = False
+
+    logFileName = 'omsconfig.log'
+    logFilePath = join(helperlib.PYTHON_PID_DIR, logFileName)
+
+    logFileHandle = open(logFilePath)
```

Comment:
>logFilePath [](start = 25, length = 11)

log file size can be as big as 50 MB. searching log line by line will be very expensive and it will reduce dsc performance.